### ISSUE TITLE
ebay-section-title: Section title CTA element not focusable by keyboard navigation

### ISSUE THREAD

**Issue by thiagobitencourt** [eBay/ebayui-core#2427]
# Bug Report

## Description

Element `section-title__cta` has properties `tabindex='-1' aria-hidden='true'` which means that's explicitly set to not support keyboard navigation.
The expected behavior is that all actionable elements should be focusable with a keyboard and screen reader so that the user can interact with them.

@agliga mentioned the following about this bug: 
> It used to have the whole title as a link and the cta was mearly decorative. That seems to be changed now

This is the code with issue: [Code Link](https://github.com/eBay/ebayui-core/blob/master/src/components/ebay-section-title/index.marko#L62)
Reproducible in Storybook as well: [Link](https://opensource.ebay.com/ebayui-core/?path=/docs/navigation-disclosure-ebay-section-title--documentation)

## Screenshots

![Image](https://github.com/user-attachments/assets/62a182b8-bbe8-4f1e-a369-43bd977b4991)
